python_modules/dagster/dagster/components/component/component.py

```diff
@@ -15,6 +17,27 @@
     from dagster.components.core.context import ComponentLoadContext
 
 
+@dataclass
+class ComponentRequirements:
+    env: dict[str, set[Path]]
+
+    def __or__(self, other: "ComponentRequirements") -> "ComponentRequirements":
+        return ComponentRequirements(
+            env={
+                key: (self.env.get(key, set()) | other.env.get(key, set()))
+                for key in set(self.env.keys()) | set(other.env.keys())
+            }
+        )
```

Comment:
would rather not use operator overloading here as it is not a primitive-esque thing. Let's just use a method